tests/testApplyLookupTable.py

```diff
@@ -0,0 +1,84 @@
+import unittest
+
+import numpy as np
+
+import lsst.utils.tests
+import lsst.afw.geom as afwGeom
+import lsst.afw.image as afwImage
+from lsst.ip.isr import applyLookupTable
+
+def referenceApply(image, table, indOffset):
+    """!Reference implementation of applyLookupTable
+
+    The algorithm is as follows:
+        numOutOfRange = 0
+        For each i,j of the image:
+            lookupInd = int(indOffset + image[i,j])
+            if lookupInd not in range [0, table.size() - 1]:
+                set lookupInd to nearest edge and increment numOutOfRange
+            image[i,j] += table[lookupInd]
+        return numOutOfRange
+
+    @param[in,out] image  image to which to add the values; modified in place
+    @param[in] table  lookup table
+    @param[in] indOffset  scalar added to image value before truncating to lookup column
+
+    @return the number of pixels whose values were out of range
+    """
+    imArr = image.getArray()
+    indArr = np.array(imArr + indOffset, dtype=int)
+    maxInd = len(table) - 1
+    numBadPoints = np.sum(indArr < 0)
+    numBadPoints += np.sum(indArr > maxInd)
+    indArr = np.where(indArr < 0, 0, indArr)
+    indArr = np.where(indArr >= maxInd, maxInd, indArr)
+    imArr += table[indArr]
+    return numBadPoints
+
+class ApplyLookupTableTestCase(lsst.utils.tests.TestCase):
+    def testBasics(self):
+        """!Test basic functionality of applyLookupTable
+        """
+        bbox = afwGeom.Box2I(afwGeom.Point2I(-31, 22), afwGeom.Extent2I(100, 85))
+        imMin = -5
+        imMax = 2500
+        tableLen = 2000
+        tableSigma = 55
+        for indOffset in (0, -50, 234):
+            for imageClass in (afwImage.ImageF, afwImage.ImageD):
+                inImage = makeRampImage(bbox=bbox, minVal=imMin, maxVal=imMax, imageClass=imageClass)
+                table = np.random.normal(scale=tableSigma, size=tableLen)
+                table = np.array(table, dtype=inImage.getArray().dtype)
+
+                refImage = imageClass(inImage, True)
+                refNumBad = referenceApply(image=refImage, table=table, indOffset=indOffset)
+
+                measImage = imageClass(inImage, True)
+                measNumBad = applyLookupTable(measImage, table, indOffset)
+
+                self.assertEqual(refNumBad, measNumBad)
+                self.assertImagesNearlyEqual(refImage, measImage)
```

Comment:
At the moment, you're testing that one implementation is the same as the "reference implementation".  But the "reference implementation" hasn't been validated, and so *both* implementations are not *known* to be free of bugs like off-by-one errors.  I think you need to add some tests that explicitly check that the right element from the table is being used, e.g., by having everything in the table being zero except for one element, and then the (single-pixel?) image has values around that corresponding to the non-zero entry in the table.